frontend_lib/src/tinymceAutoCompleteHelper.js

```diff
@@ -93,36 +94,32 @@ export const tinymceAutoCompleteHandleKeyDown = (event, setState, isAutoComplete
 }
 
 export const tinymceAutoCompleteHandleClickItem = (item, setState) => {
-  tinymce.activeEditor.focus()
-  const selection = tinymce.activeEditor.dom.select('span#autocomplete')[0]
-  tinymce.activeEditor.selection.select(tinymce.activeEditor.dom.select('span#autocomplete')[0])
-  tinymce.activeEditor.execCommand('mceInsertContent', false, `${item.mention}&nbsp;`)
-  tinymce.activeEditor.dom.remove(selection)
-
-  setState({ isAutoCompleteActivated: false })
-}
-
-export const tinymceAutoCompleteHandleSelectionChange = (selectionId, setState, isAutoCompleteActivated) => {
-  if (selectionId === 'autocomplete__searchtext' || selectionId === 'autocomplete__start') {
-    if (!isAutoCompleteActivated) setState({ isAutoCompleteActivated: true })
+  if (!item.mention) {
+    console.log('BUG: this member does not have a username')
     return
   }
 
-  if (isAutoCompleteActivated) setState({ isAutoCompleteActivated: false })
-}
+  const sel = tinymce.activeEditor.selection.getSel()
+  const text = sel.anchorNode.textContent
+
+  const posAt = text.lastIndexOf('@', sel.anchorOffset - 1)
+
+  let textBegin, textEnd
 
-export const tinymceRemoveAllAutocompleteSpan = () => {
-  let autocomplete = tinymce.activeEditor.dom.select('span#autocomplete')[0]
-  while (autocomplete) {
-    tinymce.activeEditor.focus()
-    const query = tinymce.activeEditor.getDoc().getElementById('autocomplete__searchtext').textContent.replace('@', '')
-    const selection = autocomplete
-    tinymce.activeEditor.dom.remove(selection)
-    tinymce.activeEditor.selection.select(tinymce.activeEditor.dom.select('span#autocomplete')[0])
-    tinymce.activeEditor.execCommand('mceInsertContent', false, query)
-
-    autocomplete = tinymce.activeEditor.dom.select('span#autocomplete')[0]
+  if (posAt > -1) {
+    textBegin = text.substring(0, posAt) + '@' + item.mention + '\u00A0'
+    textEnd = text.substring(seekUserNameEnd(text, sel.anchorOffset))
+  } else {
+    console.log('BUG: mention autocomplete: did not find "@"')
```

Comment:
This console log is on purpose? That's not something we normally do in Tracim's code.